internal/inspect/policy.go

```diff
@@ -296,19 +377,59 @@ func (pol *Policy) listImports(p *policyv1.Policy) (derivedRoleImports, variable
 	return derivedRoleImports, variableImports
 }
 
-// inspectDefinitionsAndRules inspects the definitions and rules in the given policy to find references to the derived roles and variables.
-func (pol *Policy) inspectDefinitionsAndRules(p *policyv1.Policy) (referencedDerivedRoles, referencedVariables map[string]struct{}, err error) {
+// inspectVariableDefinitions inspects the variable values to find references to the attributes.
+func (pol *Policy) inspectVariableDefinitions(p *policyv1.Policy) (referencedAttributes map[string]*responsev1.InspectPoliciesResponse_Attribute, err error) {
+	referencedAttributes = make(map[string]*responsev1.InspectPoliciesResponse_Attribute)
+	switch pt := p.PolicyType.(type) {
+	case *policyv1.Policy_DerivedRoles:
+		if pt.DerivedRoles.Variables != nil {
+			for _, expr := range pt.DerivedRoles.Variables.Local {
+				if err := pol.referencedAttributesInExpr(expr, referencedAttributes); err != nil {
+					return nil, fmt.Errorf("failed to find referenced attributes in the derived roles local variable: %w", err)
+				}
+			}
+		}
+	case *policyv1.Policy_PrincipalPolicy:
+		if pt.PrincipalPolicy.Variables != nil {
+			for _, expr := range pt.PrincipalPolicy.Variables.Local {
+				if err := pol.referencedAttributesInExpr(expr, referencedAttributes); err != nil {
+					return nil, fmt.Errorf("failed to find referenced attributes in the principal policy local variable: %w", err)
+				}
+			}
+		}
+	case *policyv1.Policy_ResourcePolicy:
+		if pt.ResourcePolicy.Variables != nil {
+			for _, expr := range pt.ResourcePolicy.Variables.Local {
+				if err := pol.referencedAttributesInExpr(expr, referencedAttributes); err != nil {
+					return nil, fmt.Errorf("failed to find referenced attributes in the resource policy local variable: %w", err)
+				}
+			}
+		}
+	}
+
+	return referencedAttributes, nil
+}
+
+// inspectDefinitionsAndRules inspects the definitions and rules in the given policy to find references to the attributes, derived roles and variables.
+func (pol *Policy) inspectDefinitionsAndRules(p *policyv1.Policy) (referencedAttributes map[string]*responsev1.InspectPoliciesResponse_Attribute, referencedDerivedRoles map[string]struct{}, referencedVariables map[string]*responsev1.InspectPoliciesResponse_Variable, err error) {
```

Comment:
The return values from this function are getting too complicated. It'd be better to create a struct to hold them and return that instead.